### ISSUE TITLE
Autoencoder Image Interpolation by Shaping the Latent Space

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#122]
*ICML 2021*

# TL;DR

- Previous latent space interpolation on autoencoders yield out-of-distribution samples illustrated by unwanted artifacts.
- Enforce discriminator-fidelity, cycle-consistency and smoothness on top of reconstruction to refine the latent space.

# Previous Works

- Additional discriminator to enforce reality of interpolated samples.

# Method

## Observation

![image](https://user-images.githubusercontent.com/46308407/133914041-0c8ab7ee-0b5c-4854-919d-ffd07b87700d.png)

- Simply interpolating in image domain results in ghosting objects and unwanted visual output.
- Interpolating in latent space however doesn't always produce desired result.

![image](https://user-images.githubusercontent.com/46308407/133914056-43858ec4-f892-4c68-b672-d09845bf1b39.png)

- When sampled from the autoencoder manifold (point A), the output is OK, but when sampled from out-of-manifold region, output contains irrelevant artifact (point B)

![image](https://user-images.githubusercontent.com/46308407/133914076-4aad63b8-1f23-453a-9635-97ba6a98b6dc.png)

- Combine four losses to ensure that (1) interpolation produces realistic sample (2) latent space is bijective and smooth.

## Objective

- Reconstruction
![image](https://user-images.githubusercontent.com/46308407/133914096-02879de6-e6b6-4eba-a95a-27d32ab61c31.png)

- Adversarial
![image](https://user-images.githubusercontent.com/46308407/133914098-898ea4e7-5875-411f-a2ef-7b1534e23529.png)

- Cycle-consistency in latent space
![image](https://user-images.githubusercontent.com/46308407/133914100-3a978043-2052-44b3-9d91-dc2aad4cdc2d.png)

- Smoothness w.r.t. alpha
![image](https://user-images.githubusercontent.com/46308407/133914109-1cebfe66-c699-4220-8857-300eba859eff.png)

## Overall Pipeline
![image](https://user-images.githubusercontent.com/46308407/133914120-6db417a1-a050-46e7-a728-4626b023de5a.png)

# Experiments
![image](https://user-images.githubusercontent.com/46308407/133914135-309c20f3-031f-4d82-933e-0b772931e003.png)
- Ours (AEAI) produces the best output.

![image](https://user-images.githubusercontent.com/46308407/133914144-f108d8d7-bc49-4b94-8be7-cc523ff7940a.png)
- MSE is lowest across alpha values.

![image](https://user-images.githubusercontent.com/46308407/133914149-04180843-e6f6-480c-b6be-ba5abffb5453.png)
- It predicts the GT parameter (camera elevation + azimuth) the best in synthetic dataset.

# Comments

- Tackles simple task in simple method. Good for the authors.
- Study on latent space geometry seems to be a promising area. Others tackled the problem of disentangled representation learning in terms of latent geometry and perceptual distance.
